src/components/Avatar/index.tsx

```diff
@@ -13,27 +13,58 @@ const DARK_COLORS = [
   'red',
   'purple',
   'blue',
-];
-export const RADII = ['square', 'circle'];
-export const BG_COLORS = [...LIGHT_COLORS, ...DARK_COLORS];
-export const SIZES = ['s', 'm', 'l'];
-export const INITIALS_LENGTH = [1, 2];
+] as const;
+export const RADII = ['square', 'circle'] as const;
+export const BG_COLORS = [...LIGHT_COLORS, ...DARK_COLORS] as const;
+export const SIZES = ['s', 'm', 'l'] as const;
+export const INITIALS_LENGTH = [1, 2] as const;
 
-export const grabInitials = (str, initialsCount = 2) =>
+export const grabInitials = (str: string, initialsCount = 2): string =>
   str
     .split(' ')
     .slice(0, initialsCount)
     .reduce((prev, curr) => prev + curr.charAt(0), '')
     .toUpperCase();
 
-export const trimName = (str) => {
+export const trimName = (str: string): string | null => {
   // We want to reject strings which only contain whitespaces
   const regex = /\S/;
   // If it doesn't only contain whitespaces, return a trimmed string,
   // otherwise return null
   return str && regex.test(str) ? str.trim() : null;
 };
 
+export interface AvatarProps
+  extends HTMLAttributes<HTMLElement | HTMLButtonElement | HTMLAnchorElement> {
+  /** Controls color of button */
+  bgColor?: typeof BG_COLORS[number];
+  /** Controls the size of the button */
+  size?: typeof SIZES[number];
+  /** Controls radius of button (slightly rounded square, or circle) */
+  radius?: typeof RADII[number];
+  /** Control initials displayed, and also used for title accessibility attribute (uses 2 words max for initials) */
+  name?: string;
+  /** Sets background image over initials */
+  imgUrl?: string;
+  /** Controls how many initials can be displayed */
+  initialsLength?: typeof INITIALS_LENGTH[number];
+  /**  Accepts FDS Icon. Renders if name or imgUrl are not present  */
+  PlaceholderIcon?: Icon;
+  /**
+   * Takes in a react-router `Link` reference and sets it
+   * as the base element. You may ONLY use it like the
+   * following:
+   *
+   * - `import { Link } from 'react-router'`
+   * - `Link={Link}`
+   */
+  Link?: React.ElementType;
```

Comment:
[Could] This type could be made more narrow by using `React.ComponentType`. `React.ElementType` will accept values like `'a'` and `'div'` since they are valid elements, but they are not components. In this case `React.ComponentType` is a subset of `React.ElementType`.

Reply:
Fair enough

Reply:
✅